### PR TITLE
[master] update pre-commit-config

### PR DESCRIPTION
**Pull request by cmcmarrow**
### What does this PR do?
update pre-commit-config

### Commits signed with GPG?
Yes

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ repos:
             )$
 
   - repo: https://github.com/psf/black
-    rev: 22.6.0
+    rev: 23.9.1
```
[the remark this answers is not in the document]
Comment (s0undt3ch) [reply .pre-commit-config.yaml:1253]: https://github.com/saltstack/salt/pull/65314/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9R1272 make it match

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Comment by s0undt3ch** [comment]
Let's also make these changes please:
```diff
diff --git a/.github/workflows/ci.yml b/.github/workflows/ci.yml
index f9af175715..6d123775a8 100644
--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -128,6 +128,9 @@ jobs:
                 - *salt_added_modified
                 - *tests_added_modified
                 - *pkg_tests_added_modified
+            pre-commit:
+              - added|modified:
+                - .pre-commit-config.yaml

       - name: Set up Python 3.10
         uses: actions/setup-python@v4
diff --git a/.github/workflows/pre-commit-action.yml b/.github/workflows/pre-commit-action.yml
index fbafa2c4b4..cf0e332b0d 100644
--- a/.github/workflows/pre-commit-action.yml
+++ b/.github/workflows/pre-commit-action.yml
@@ -50,14 +50,14 @@ jobs:
           cache-seed: ${{ inputs.cache-seed }}

       - name: Check ALL Files On Branch
-        if: github.event_name != 'pull_request'
+        if: github.event_name != 'pull_request' || fromJSON(inputs.changed-files)['pre-commit']
         env:
           SKIP: lint-salt,lint-tests,remove-import-headers,rstcheck
         run: |
           pre-commit run --show-diff-on-failure --color=always --all-files

       - name: Check Changed Files On PR
-        if: github.event_name == 'pull_request' && fromJSON(inputs.changed-files)['repo']
+        if: github.event_name == 'pull_request' && fromJSON(inputs.changed-files)['repo'] && fromJSON(inputs.changed-files)['pre-commit'] == false
         env:
           SKIP: lint-salt,lint-tests
           GH_ACTIONS_ANNOTATE: "1"
```
We want to run pre-commit on everything when `.pre-commit-config.yaml` changes

**Comment by s0undt3ch** [comment]
Sorry

```diff
diff --git a/.github/workflows/pre-commit-action.yml b/.github/workflows/pre-commit-action.yml
index fbafa2c4b4..cf0e332b0d 100644
--- a/.github/workflows/pre-commit-action.yml
+++ b/.github/workflows/pre-commit-action.yml
@@ -50,14 +50,14 @@ jobs:
           cache-seed: ${{ inputs.cache-seed }}

       - name: Check ALL Files On Branch
-        if: github.event_name != 'pull_request'
+        if: github.event_name != 'pull_request' || fromJSON(inputs.changed-files)['pre-commit']
         env:
           SKIP: lint-salt,lint-tests,remove-import-headers,rstcheck
         run: |
           pre-commit run --show-diff-on-failure --color=always --all-files

       - name: Check Changed Files On PR
-        if: github.event_name == 'pull_request' && fromJSON(inputs.changed-files)['repo']
+        if: github.event_name == 'pull_request' && fromJSON(inputs.changed-files)['repo'] && fromJSON(inputs.changed-files)['pre-commit'] == false
         env:
           SKIP: lint-salt,lint-tests
           GH_ACTIONS_ANNOTATE: "1"
diff --git a/.github/workflows/templates/layout.yml.jinja b/.github/workflows/templates/layout.yml.jinja
index 773ebeb622..421b2f7ec1 100644
--- a/.github/workflows/templates/layout.yml.jinja
+++ b/.github/workflows/templates/layout.yml.jinja
@@ -176,6 +176,9 @@ jobs:
                 - *salt_added_modified
                 - *tests_added_modified
                 - *pkg_tests_added_modified
+            pre-commit:
+              - added|modified:
+                - .pre-commit-config.yaml
```

**Comment by cmcmarrow** [comment]
Thansk @s0undt3ch

**Comment by s0undt3ch** [comment]
As you can see, if we upgrade `black` there's a lot of code that needs to change in Salt, and, if the team decides to do that, we should also consider running `pyupgrade` (`pre-commit run -av pyupgrade`) against all of the Salt code base.

**Review by twangboy — APPROVED** [review]

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**